airflow/settings.py

```diff
@@ -228,6 +229,19 @@ def configure_vars():
     global PLUGINS_FOLDER
     global DONOT_MODIFY_HANDLERS
     SQL_ALCHEMY_CONN = conf.get('core', 'SQL_ALCHEMY_CONN')
+
+    # as of sqlalchemy 1.4, scheme `postgres+psycopg2` must be replaced with `postgresql`
+    parsed = urlparse(SQL_ALCHEMY_CONN)
+    bad_scheme = 'postgres+psycopg2'
+    if parsed.scheme == bad_scheme:
+        warnings.warn(
+            f"Scheme for metadata sql alchemy connection is `{bad_scheme}`."
+            "As of sqlalchemy 1.4 this is no longer supported.  You must change "
+            "to `postgresql`",
+            PendingDeprecationWarning,
```

Comment:
Should this be `DeprecationWarning`?

Reply:
That's a good point. I think there is no real difference between PendingDeprecationWarning and DeprecationWarning and seems that it's semantics and meaning has been changed in the way to make PendingDeprecationWarning - well -  Deprecated :) 

https://docs.python.org/3/library/exceptions.html#PendingDeprecationWarning

> This class is rarely used as emitting a warning about a possible upcoming deprecation is unusual, and [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) is preferred for already active deprecations.

Should we change it everywhere in our code ? (or does it need a deprecation to change PendingDeprecationWarning into DeprecationWarning 🤣  ?

Reply:
We probably should.

By the way sqlalchemy does something really nice where it makes it's own deprecation warnning with specific class
https://github.com/sqlalchemy/sqlalchemy/blob/769fa67d842035dd852ab8b6a26ea3f110a51131/lib/sqlalchemy/exc.py#L779

for example:
`[2022-03-17 09:02:03,734] {logging_mixin.py:115} WARNING - /opt/***/***/models/dag.py:1084 SADeprecationWarning: Query.value() is deprecated and will be removed in a future release.  Please use Query.with_entities() in combination with Query.scalar() (deprecated since: 1.4)`

It's very nice to know what package originated the warning.

Reply:
> or does it need a deprecation to change PendingDeprecationWarning into DeprecationWarning

yes, absolutely :trollface: